app/src/main/java/com/islaharper/dawnofandroid/di/NetworkModule.kt

```diff
@@ -52,7 +54,13 @@ object NetworkModule {
 
     @Provides
     @Singleton
-    fun provideAuthApi(retrofit: Retrofit): MyApi {
+    fun provideApi(retrofit: Retrofit): MyApi {
         return retrofit.create(MyApi::class.java)
     }
+
+    @Provides
+    @Singleton
+    fun provideRemoteDataSource(myApi: MyApi): RemoteDataSource {
```

Comment:
Minor:
Please use `@Inject` on the constructor wherever possible and use `@Binds` to associate an interface with an implementation. Not only does this save us writing code, the internally generated code from Dagger is more efficient (according to the docs)

```kotlin
@Inject
class RemoteDataSourceImpl @Inject constructor(private val myApi: MyApi)

@Inject
abstract fun Module {
    @Binds
    abstract fun bindRemoteDataSource(remoteDataSourceImpl: RemoteDataSourceImpl): RemoteDataSource
}
```

https://dagger.dev/dev-guide/

Reply:
Thanks, I'm still quite new to Dagger/Hilt, DI in general, so this is super helpful. Will make the changes in the next commit.